frontend/src/features/public-form/components/FormFields/VisibleFormFields.tsx

```diff
@@ -47,7 +49,17 @@ export const VisibleFormFields = ({
     )
     const visibleFieldsWithQuestionNo = augmentWithQuestionNo(visibleFields)
     setVisibleFormFields(visibleFieldsWithQuestionNo)
-  }, [formFields, formLogics, setVisibleFieldIdsForScrollData, watchedValues])
+
+    // set the number of visible fields in the context for public forms
+    if (setNumVisibleFields)
```

Comment:
Yep! For Preview mode it will be `undefined`! Cause we aren't sending any `responseMetadata` on preview mode responses.

I do agree its a bit weird to use a setter as a conditional 😅 was trying to tie it to `isPreview` but couldn't really get the typing to work.

Reply:
Ooh huh, okay lemme take a closer look

Reply:
Okay yea I think because we need to explicitly define 2 types for preview form and public form for this to work cleanly. Okay just use the setter as a conditional I guess, can you add a comment for this?

Reply:
[c420ccb](https://github.com/opengovsg/FormSG/pull/6377/commits/c420ccb3e5efe22acc5b580925557220caabd873)

Thanks HQ!